cmd/katapult/client_test.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/krystal/go-katapult"
+
+	"github.com/krystal/katapult-cli/config"
+)
+
+func TestNewClient(t *testing.T) {
+	tests := []struct {
+		name string
+
+		apiKey string
+		apiURL string
+	}{
+		{
+			name:   "Handle blank API URL",
+			apiKey: "test",
+		},
+		{
+			name:   "Handle both values",
+			apiKey: "test",
+			apiURL: "https://example.com",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c, err := newClient(&config.Config{
+				APIKey: tt.apiKey,
+				APIURL: tt.apiURL,
+			})
+			require.NoError(t, err)
```

Comment:
As `newClient` can return an error, and has a couple of error specific cases of logic to handle errors, I think we should probably have some test cases that cover the possible error conditions.